frontend/src/composables/useRequestFee.ts

```diff
@@ -32,11 +38,24 @@ export function useRequestFee(
       console.error(errorMessage);
       error.value = errorMessage ?? 'Unknown failure.';
     }
+
+    loading.value = false;
   };
 
-  watch([rpcUrl, requestManagerAddress, requestAmount], updateRequestFeeAmount, {
-    immediate: true,
-  });
+  const cb = computed(() =>
+    debounced ? useDebouncedTask(updateRequestFeeAmount, debouncedDelay) : updateRequestFeeAmount,
+  );
+
+  watch(
+    [rpcUrl, requestManagerAddress, requestAmount],
+    () => {
+      loading.value = true;
+      cb.value();
+    },
+    {
+      immediate: true,
```

Comment:
I was wondering if there is an easy way to not have the spinner on page load. Maybe we don't need this immediate option and it would work?

Reply:
I just checked this. 
Removing the `immediate: true` option would not change the behavior since the `rpcUrl` and `requestManagerAddress` are changing on page load. This happens only when user is connected to a network in his wallet that we support inside the sourceChain selector. In that case, the app prefills the sourceChain selection input with his currently connected address and therefore those values change and the `loading` value is set to true, even tho not needed when amount is still `undefined`.

However, i can solve this by adding a simple condition inside the watcher function which checks if the token amount is empty. If so, set the fee amount to `undefined` in that case.

Reply:
Go for it if you want to. It's really not a requirement, just looks a littler nicer to have a stable page on load I think.